src/jit/morph.cpp

```diff
@@ -12503,31 +12503,44 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
 
             /* See if we can fold GT_ADD nodes. */
 
-            if (oper == GT_ADD)
+            if (oper == GT_ADD || oper == GT_OR)
             {
-                /* Fold "((x+icon1)+(y+icon2)) to ((x+y)+(icon1+icon2))" */
+                /* Fold "((x+icon1)+(y+icon2)) to ((x+y)+(icon1+icon2))" same for bitwise OR */
```

Comment:
The `same for bitwise OR` is not really very descriptive. Probably would be better to say something like:
```
// Fold "(x <op> icon1) <op> (y <op> icon2))" to "((x <op> y) <op> (icon1 <op> icon2))"